src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -1138,6 +1145,33 @@ public int GetCurrentExceptionSpanPosition(uint id, VsTextSpan[] ptsNewPosition)
             }
         }
 
+        private static int MarshalPdbReader(IDebugUpdateInMemoryPE2 updater, out IntPtr pdbReaderPointer)
+        {
+            // ISymUnmanagedReader can only be accessed from an MTA thread, however, we need
+            // fetch the IUnknown instance (call IENCSymbolReaderProvider.GetSymbolReader) here
+            // in the STA.  To further complicate things, we need to return synchronously from
+            // this method.  Waiting for the MTA thread to complete so we can return synchronously
+            // blocks the STA thread, so we need to make sure the CLR doesn't try to marshal
+            // ISymUnmanagedReader calls made in an MTA back to the STA for execution (if this
+            // happens we'll be deadlocked).  We'll use CoMarshalInterThreadInterfaceInStream to
+            // achieve this.  First, we'll marshal the object in a Stream and pass a Stream pointer
+            // over to the MTA.  In the MTA, we'll get the Stream from the pointer and unmarshal
+            // the object.  I *believe* the reader object was originally created on an MTA thread,
```

Comment:
You can remove the "I *believe*" bit. I got confirmation from GreggM that this is the case.